plugins/out_s3/s3.c

```diff
@@ -1540,7 +1547,14 @@ static struct multipart_upload *create_upload(struct flb_s3 *ctx,
     m_upload->tag = tmp_sds;
     m_upload->upload_state = MULTIPART_UPLOAD_STATE_NOT_CREATED;
     m_upload->part_number = 1;
-    m_upload->init_time = time(NULL);
+    if (tms == NULL) {
+        m_upload->init_time = time(NULL);
+        flb_plg_error(ctx->ins, "Failed to get timestamp from the log record "
+                      "used for the S3 key, use the current time instead.");
+    }
+    else {
+        m_upload->init_time = tms->tm.tv_sec;
+    }
```

Comment:
@Claych Further up in this code we create the S3 key, this is where the init_time timestamp needs to be used... right now we are still using time(NULL)... please address this and test it.